app/src/main/java/com/tiptopgoodstudio/androidresources/ui/MainActivity.java

```diff
@@ -2,132 +2,85 @@
-    private TextView mErrorMessage;
+    private StickyNoteAdapter mStickyNoteAdapter;
+    private View mSplashScreen;
+    private Toolbar mToolbar;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-
         setContentView(R.layout.activity_main);
 
-        Toolbar toolbar = (Toolbar) findViewById(R.id.action_toolbar_main);
-        toolbar.setBackgroundColor(getResources().getColor(R.color.colorPrimary));
-        toolbar.setTitleTextColor(getResources().getColor(R.color.colorMenuItem));
-        setSupportActionBar(toolbar);
+        initializeUiViews();
+        setupActivityAppBar();
+        setupRecyclerView();
+        setupActivityAsDataObserver();
+    }
 
-        mRecyclerView = (RecyclerView) findViewById(R.id.rv_stickynotes);
-        mLoadingIndicator = (ProgressBar) findViewById(R.id.pb_loading_indicator);
-        mErrorMessage = (TextView) findViewById(R.id.main_error_message_display);
+    private void initializeUiViews() {
+        mToolbar = findViewById(R.id.action_toolbar_main);
+        mRecyclerView = findViewById(R.id.rv_stickynotes);
+        mSplashScreen = findViewById(R.id.splash_screen);
+    }
 
+    private void setupRecyclerView() {
         GridLayoutManager layoutManager = new GridLayoutManager(MainActivity.this, 2, GridLayoutManager.VERTICAL, false);
         mRecyclerView.setLayoutManager(layoutManager);
+        mStickyNoteAdapter = new StickyNoteAdapter(this);
+        mRecyclerView.setAdapter(mStickyNoteAdapter);
+    }
 
-        mSitckyNoteAdapter = new SitckyNoteAdapter(this);
-        mRecyclerView.setAdapter(mSitckyNoteAdapter);
-
-        // Load the topics list from the ViewModel
-        loadDataFromViewModel();
+    private void setupActivityAppBar() {
+        mToolbar.setBackgroundColor(getResources().getColor(R.color.colorPrimary));
+        mToolbar.setTitleTextColor(getResources().getColor(R.color.colorMenuItem));
+        setSupportActionBar(mToolbar);
     }
 
     /**
      * This method retrieves data from the ViewModel
-     *
-     * Added by Divya on 4/21/2018
      */
-    private void loadDataFromViewModel(){
-
-        try {
-
-            showProgress();
-
-            TopicViewModel model = ViewModelProviders.of(this).get(TopicViewModel.class);
-
-            model.getTopicsList().observe(this, topicList -> {
-                //Update UI
-                mSitckyNoteAdapter.setTopicData(topicList);
+    private void setupActivityAsDataObserver() {
+
+       TopicViewModel mTopicViewModel = ViewModelProviders.of(this).get(TopicViewModel.class);
+
+        mTopicViewModel.getTopicsList().observe(this, topicList -> {
+            //Update UI
+            mStickyNoteAdapter.setTopicData(topicList);
+            if (mStickyNoteAdapter.getItemCount() == 0) {
+                if (Utilities.isOnline(this)) {
+                    showSplashScreen();
+                } else {
+                    showSplashScreen();
```

Comment:
If the showSplashScreen() is called in the if block and the else block, then it can be moved out of the if else block

Reply:
Good catch!  I fixed.